src/gpuarray_reduction.c

```diff
@@ -188,63 +695,101 @@ static void  appendIdxes                        (strb*              s,
 
 /**
  * @brief Check the sanity of the arguments, in agreement with the
- *        documentation for GpuArray_maxandargmax().
+ *        documentation for GpuArray_reduction().
  *
  *        Also initialize certain parts of the context.
  *
  * @return GA_INVALID_ERROR if arguments invalid; GA_NO_ERROR otherwise.
  */
 
-static int   maxandargmaxCheckargs              (maxandargmax_ctx*  ctx){
-	int i;
+static int   reduxCheckargs                (redux_ctx*  ctx){
+	int i, ret;
+	const strb INIT_STRB = STRB_STATIC_INIT;
 
 	/**
 	 * We initialize certain parts of the context.
 	 */
 
-	ctx->ret           = GA_NO_ERROR;
 	ctx->axisList      = NULL;
 	ctx->gpuCtx        = NULL;
 
-	ctx->dstMaxType    = ctx->dstArgmaxType = NULL;
+	ctx->srcTypeStr    = ctx->dstTypeStr    = ctx->dstArgTypeStr =
+	ctx->accTypeStr    = ctx->idxTypeStr    = NULL;
+	ctx->initVal       = NULL;
 	ctx->ndh           = 0;
+	ctx->ndhd          = 0;
+	ctx->ndhr          = 0;
 	ctx->sourceCode    = NULL;
+	ctx->s             = INIT_STRB;
```

Comment:
The problem with that idea is that it's not future-proof. I have no reason to believe that `memset`ing `strb`'s to zero will always be a correct way of doing it. In C, the typical way this would have been resolved is to offer an `strb_init(strb*)` function; In fact that solution is more popular than static initializers. The current static initializer is currently the only reliable way of doing this, because `strb_clear(strb*)` presumes that the `strb` in question has been initialized in the past (by dint of the fact it unconditionally `free`s a pointer).